opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceCreation.jsx

```diff
@@ -46,12 +71,36 @@ const sharedUpdater = (store, userId, paginationOptions, newEdge) => {
   ConnectionHandler.insertEdgeBefore(conn, newEdge);
 };
 
-class WorkspaceCreation extends Component {
-  onSubmit(values, { setSubmitting, resetForm }) {
+const WorkspaceCreation = ({ paginationOptions, type }) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+  const inputRef = useRef();
+  const [commitImportMutation] = useMutation(importMutation);
+  const history = useHistory();
+
+  const handleImport = (event) => {
+    const importedFile = event.target.files[0];
+    commitImportMutation({
+      variables: { file: importedFile },
+      onCompleted: (data) => {
+        history.push(
+          `${resolveLink('Dashboard')}/${data.configurationImport}`,
+        );
+      },
+      onError: (error) => {
+        handleError(error);
+      },
+    });
+  };
+
+  const onSubmit = (values, { setSubmitting, resetForm }) => {
     commitMutation({
```

Comment:
sorry I just saw this when reviewing again, but since you refactored this component, we should now use `useMutation` as you did with `importMutation` instead of `commitMutation` for `workspaceMutation`

Reply:
Or maybe it was not possible to use it ?

Reply:
Let me check this out